### ISSUE TITLE
swarmsim.com

### ISSUE THREAD

**Issue by erosson** [erosson/swarm#119]
I bought this on a whim the day I first made this thing public. let's use it instead of github.io.

**Comment by erosson** [comment]
swarmsim.github.io inspects progress on swarmsim-publictest.github.io's via iframe and window.postMessage() to award an achievement, #267. we've had exactly 0 reports of this mechanism breaking, despite my fears about browser extensions/firewalls/browser incompatibility/etc. screwing with the cross-domain communication. a seamless migration to swarmsim.com might be practical after all!

main blocker now is ssl; we have ssl now so I'd rather not drop it. https://letsencrypt.org/ will offer multiple free prod-worthy ssl certs in the near-ish future, and I'm in no hurry and a total cheapskate, so let's just wait.

doesn't look like traffic will force us into rushing away from github hosting early: https://www.quora.com/What-are-bandwidth-and-traffic-limits-for-GitHub-pages

> We don't have any hard limits. Every now and then a massive porn site hotlinks a JavaScript file or something and we have to ban them, but it's so incredibly rare that you'll ever run into any problems.

**Comment by Equinoxdawg** [comment]
I just thought I'd give my opinion on this... the SSL isn't really needed, but I completely understand and think the same as you do, it's nice to have it regardless. I recently went through trying to get a SSL cert for my site, and I found https://letsencrypt.org/ too- which if I remember said it would be available mid-2015. It's definitely worth the wait, especially since there are some pretty heavy hitters behind it: Mozilla, namely. I'd recommend to wait for that... in the meantime, you could opt for CloudFlare's free plan, which gives you an SSL (it encrypts the connection from the viewer to their servers, but not their servers to yours - the full encryption requires a cert from somewhere) - which doesn't really matter since you really don't have any personal information that people need to worry about. I'd reccomend to look at https://www.cloudflare.com/plans - it's pretty robust and you might even like the optomisations it offers - although I'm fairly sure you can turn down everything other than just the SSL cert. And of course you could try it with a test site and see how that goes, then decide - the lovely part about it being free without any obligations.

**Comment by erosson** [comment]
I had no idea cloudflare had a free plan; nifty. Signed up, trying them out.

Thanks for the pointers!

**Comment by erosson** [comment]
wow, this worked! Cloudflare does indeed have free ssl, and it works well enough for my needs with github pages. staging environment ( https://erosson.github.io/swarm/ ): now https://test.swarmsim.com/swarm/ - should remove the trailing directory from that now, but it's a working proof of concept!

I understand the cloudflare <-> github connection isn't encrypted, but like you said, for swarmsim's case that's fine - we're just static files with no sensitive data.

@Equinoxdawg, thank you for your help on this!

so now I know how to do this, just need to make the time for it. other stuff's more important so this is probably still a while off, but now it's not blocked on anything.

**Comment by Equinoxdawg** [comment]
I'm glad it worked out!
